### ISSUE TITLE
Nunit.Analyzers not applying code fixes

### ISSUE THREAD

**Issue by joshboyceplexure** [nunit/nunit.analyzers#705]
I have a couple of issues:

Firstly, I only sometimes receive warnings for converting classic asserts to constraint models. Cleaning & rebuilding solution only sometimes works. When it does work, it only shows warnings for a handful of files & not all files with Assert in the solution

Secondly, the warning that I do get does not suggest the code fix to convert from classic to constraint models. So currently I would have to update ~1000 lines of code manually

IDE: VS Enterprise 2022 v17.9.0
Nunit 3.14.0
Nunit.Analyzers 4.0.1

**Comment by stevenaw** [comment]
@joshboyceplexure Do you have an example snippet you can share to help the team reproduce your issue?

**Comment by manfred-brands** [comment]
As the conversions are "suggestions", VS doesn't raise them unless you have the file open. Even when run analyzer on whole solution. I have the same problem for issues raised by any analyzer.

There are configuration options under Text Editor -> C# -> Advanced:
![image](https://github.com/nunit/nunit.analyzers/assets/8655218/eb000a1f-dba3-4d30-a5ab-93c4b89558ef)

By default it only runs analyzers on the current document and even only shows warnings on open documents.

Normally you wouldn't want VS to run analysis on the whole solution as it would slow you down when you only work on a subset, except for complete solution refactoring. 

You can try to change these settings.

What also works though is when you have a CodeFix suggestion, then do an "Fix all occurrences in solution":
![image](https://github.com/nunit/nunit.analyzers/assets/8655218/e0e78528-b575-4942-be9c-4426edbfe61a)

Also to ensure you didn't miss any, you can change the severity of these rules to error:
```
# Default severity for analyzer diagnostics with category 'Assertion'
dotnet_analyzer_diagnostic.category-Assertion.severity = error
```

**Comment by joshboyceplexure** [comment]
@manfred-brands thank you for your suggestions! By applying the change to run the analyser on the whole solution it also seemed to slowly show the CodeFix suggestions as well. Maybe because my project is quite large it was taking too long to show the fixes in the code itself.

Setting the severity to error level also helped catch some fixes I missed previously

**Comment by dragos-durlut** [comment]
Another reason the `CodeFix` suggestions do not appear is because you might not have the `NUnit.Analyzers` package added in the project file.

**Comment by joshboyceplexure** [comment]
Actually, I found another cause of the issue: Some of the files had `using Microsoft.VisualStudio.TestTools.UnitTesting` instead of `using NUnit.Framework`. This meant that the suggestions never showed up on those files & went unnoticed

**Comment by OsirisTerje** [comment]
@joshboyceplexure How did that compile?

**Comment by joshboyceplexure** [comment]
@OsirisTerje The Microsoft package also has Assert functions identical to Nunit
![image](https://github.com/nunit/nunit.analyzers/assets/158102314/8a6f2a86-90f3-40c4-97a5-cd494be57010)

**Comment by OsirisTerje** [comment]
@joshboyceplexure Yes, but the test method has to be marked as a [Test] in NUnit and [TestMethod] in MSTest.  And using the NUnit attribute when having a using to MSTest should give a compile error.

**Comment by lightng** [comment]
I have also seen this problem when there is "Microsoft.NET.Test.Sdk" in the package references. In this case removing it solved the issue for me.
